src/runtime_src/driver/zynq/drm/zocl/sched_exec.c

```diff
@@ -546,9 +529,30 @@ configure(struct sched_cmd *cmd)
 	}
 
 	for (i = 0; i < exec->num_cus; i++) {
-		exec->cu_addr_phy[i] = cfg->data[i];
-		SCHED_DEBUG("++ configure cu(%d) at 0x%x\n", i,
-		    exec->cu_addr_phy[i]);
+		/* If it is in ert mode, there is no XCLBIN parsed now
+		 * Trust configuration from host. Once host download XCLBIN to
+		 * PS side, verify host configuration in the same way.
+		 *
+		 * Now the zdev->ert is heavily used in configure()
+		 * Need cleanup.
+		 */
+		if (!zdev->ert && is_valid_apts(zdev, cfg->data[i]) < 0) {
+			write_unlock(&zdev->attr_rwlock);
+			return 1;
+		}
+		/* For MPSoC as PCIe device, the CU address for PS = base
+		 * address + PCIe offset.
+		 *
+		 * For Pure MPSoC device, the base address is always 0
+		 */
+		exec->cu_addr_phy[i] = zdev->res_start + cfg->data[i];
+		exec->cu_addr_virt[i] = ioremap(exec->cu_addr_phy[i], CU_SIZE);
+		if (!exec->cu_addr_virt[i]) {
+			write_unlock(&zdev->attr_rwlock);
```

Comment:
We may want to add DRM_ERR to indicate why the configuration fails

Reply:
done